### PR TITLE
[MXNET-1232] fix demo and add Eclipse support

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##
Get rid of makefile and add Eclipse support for Java
@andrewfayres @piyushghai @zachgk @aaronmarkham 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: docs/install/java_setup.md [hunk 1]
```diff
@@ -85,6 +85,12 @@ Also, add the dependency which corresponds to your platform to the `dependencies
 The official Java Packages will be released with the release of MXNet 1.4 and will be available on  [MXNet Maven package repository](https://search.maven.org/#search%7Cga%7C1%7Cg%3A%22org.apache.mxnet%22).
 <hr>
 
+### Eclipse IDE Support
+You can easily convert your maven project to Eclipse project by doing:
```
Comment (piyushghai) [inline docs/install/java_setup.md]: ```You can easily convert you existing maven project to a project that can run in Eclipse by```

File: scala-package/mxnet-demo/java-demo/README.md [hunk 2]
```diff
@@ -1,13 +1,13 @@
 # MXNet Java Sample Project
 This is an project created to use Maven-published Scala/Java package with two Java examples.
 ## Setup
-You can use the `Makefile` to make the Java package. Simply do the following:
-```Bash
-make javademo
+User are required to use maven to build the package, by simply run
```
Comment (piyushghai) [inline scala-package/mxnet-demo/java-demo/README.md]: run --> running

File: scala-package/mxnet-demo/java-demo/README.md [hunk 3]
```diff
@@ -71,9 +71,15 @@ If you want to test run on GPU, you can set a environment variable as follows:
...
 ## Clean up
-Clean up for Maven package is simple, you can run the pre-configed `Makefile` as:
+Clean up for Maven package is simple:
 ```Bash
-make javaclean
+mvn clean
+```
+
+## Run in Eclipse
+You can convert the maven project to the eclipse one by simply do:
```
Comment (piyushghai) [inline scala-package/mxnet-demo/java-demo/README.md]: ```You can convert the maven project to the eclipse one by running```

File: scala-package/mxnet-demo/java-demo/pom.xml [hunk 4]
```diff
@@ -21,7 +28,30 @@
...
+        <profile>
+          <id>linux-x86_64</id>
+          <activation>
+          <os>
+            <family>unix</family>
+            <name>Linux</name>
+          </os>
+          </activation>
+          <properties>
+            <mxnet.profile>linux-x86_64-cpu</mxnet.profile>
```
Comment (piyushghai) [inline scala-package/mxnet-demo/java-demo/pom.xml]: This will always pick the cpu artifact. The existing code allows one to pick up the gpu artifacts as well. 

https://github.com/apache/incubator-mxnet/blob/master/scala-package/mxnet-demo/scala-demo/Makefile#L35

File: scala-package/mxnet-demo/scala-demo/pom.xml [hunk 5]
```diff
@@ -22,10 +22,35 @@
...
+        <profile>
+          <id>linux-x86_64</id>
+          <activation>
+          <os>
+            <family>unix</family>
+            <name>Linux</name>
+          </os>
+          </activation>
+          <properties>
+            <mxnet.profile>linux-x86_64-cpu</mxnet.profile>
```
Comment (piyushghai) [inline scala-package/mxnet-demo/scala-demo/pom.xml]: Same comment here for gpu v/s cpu.

File: scala-package/mxnet-demo/java-demo/pom.xml (hunk 4 above)
Comment (lanking520) [reply scala-package/mxnet-demo/java-demo/pom.xml]: In order to simplify the process, I decided to get rid of the makefile entirely. To address your concern, I add a noteline in the README to inform then using GPU to run

Comment (piyushghai) [reply scala-package/mxnet-demo/java-demo/pom.xml]: Yeah. removing makefile was a good riddance :) 

I'm fine with adding it to the README.

**Comment by lanking520** [comment]
@aaronmarkham Could you please review this?

File: docs/install/java_setup.md [hunk 6]
```diff
@@ -85,6 +85,12 @@ Also, add the dependency which corresponds to your platform to the `dependencies
 The official Java Packages will be released with the release of MXNet 1.4 and will be available on  [MXNet Maven package repository](https://search.maven.org/#search%7Cga%7C1%7Cg%3A%22org.apache.mxnet%22).
 <hr>
 
+### Eclipse IDE Support
+You can easily convert you existing maven project to a project that can run in Eclipse by:
```
Comment (aaronmarkham) [inline docs/install/java_setup.md]:
```suggestion
You can convert your existing Maven project to a project that can run in Eclipse by:
```

File: docs/install/java_setup.md [hunk 7]
```diff
@@ -85,6 +85,12 @@ Also, add the dependency which corresponds to your platform to the `dependencies
...
+```
+mvn eclipse:eclipse
```
Comment (aaronmarkham) [inline docs/install/java_setup.md]: From what directory? Is this known or should you mention it?

File: docs/install/java_setup.md [hunk 8]
```diff
@@ -99,11 +105,11 @@ The previously mentioned setup with Maven is recommended. Otherwise, the followi
 
 
 #### Build Java from an Existing MXNet Installation
-If you have already built MXNet **from source** and are looking to setup Java from that point, you may simply run the following from the MXNet source root:
+If you have already built MXNet **from source** and are looking to setup Java from that point, you may simply run the following from the MXNet `scala-package`:
```
Comment (aaronmarkham) [inline docs/install/java_setup.md]:
```suggestion
If you have already built MXNet **from source** and are looking to setup Java from that point, you may simply run the following from the MXNet `scala-package` folder:
```

File: scala-package/mxnet-demo/java-demo/README.md [hunk 9]
```diff
@@ -1,13 +1,15 @@
 # MXNet Java Sample Project
 This is an project created to use Maven-published Scala/Java package with two Java examples.
 ## Setup
-You can use the `Makefile` to make the Java package. Simply do the following:
-```Bash
-make javademo
+User are required to use maven to build the package, by simply running
```
Comment (aaronmarkham) [inline scala-package/mxnet-demo/java-demo/README.md]:
```suggestion
You are required to use Maven to build the package with the following commands:
```
However, isn't this not true? You can run the sample project by building for source, can you not?

File: scala-package/mxnet-demo/java-demo/README.md [hunk 10]
```diff
@@ -1,13 +1,15 @@
...
+```
+mvn package
 ```
-This will load the default parameter for all the environment variable.
-If you want to run with GPU on Linux, just simply add `USE_CUDA=1` when you run the make file
- 
+This command will pick the default values specified in the pom file.
```
Comment (aaronmarkham) [inline scala-package/mxnet-demo/java-demo/README.md]: Link pom file.

File: scala-package/mxnet-demo/java-demo/README.md [hunk 11]
```diff
@@ -1,13 +1,15 @@
...
+
+Note: If you are planning to use GPU, please add `-Dmxnet.profile=linux-x86_64-gpu`
+
+### Use customized version set
 You can use the following instruction as an alternative to achieve the same result:
 User are required to use `mvn package` to build the package,
```
Comment (aaronmarkham) [inline scala-package/mxnet-demo/java-demo/README.md]: Grammar on this line... also is this true? Should it be you "may" not "must" or "required"?
```suggestion
You are required to use `mvn package` to build the package,
```

File: scala-package/mxnet-demo/java-demo/README.md [hunk 12]
```diff
@@ -1,13 +1,15 @@
...
  which are shown below:
```
Comment (aaronmarkham) [inline scala-package/mxnet-demo/java-demo/README.md]:
```suggestion
using the following commands:
```

File: scala-package/mxnet-demo/java-demo/README.md [hunk 13]
```diff
@@ -71,9 +73,15 @@ If you want to test run on GPU, you can set a environment variable as follows:
...
 ## Clean up
-Clean up for Maven package is simple, you can run the pre-configed `Makefile` as:
+Clean up for Maven package is simple:
 ```Bash
-make javaclean
+mvn clean
+```
+
+## Run in Eclipse
+You can convert the maven project to the eclipse one by running
```
Comment (aaronmarkham) [inline scala-package/mxnet-demo/java-demo/README.md]:
```suggestion
You can convert the maven project to the eclipse one by running the following command:
```

File: scala-package/mxnet-demo/java-demo/README.md [hunk 14]
```diff
@@ -71,9 +73,15 @@ If you want to test run on GPU, you can set a environment variable as follows:
...
+```
+mvn eclipse:eclipse
```
Comment (aaronmarkham) [inline scala-package/mxnet-demo/java-demo/README.md]: This is a repeat of before... why is this here?
Also this is a conversion instruction not a "run" instruction as the title suggests. Maybe you can include some screenshots of how you open the project in eclipse and build it, etc...

File: scala-package/mxnet-demo/java-demo/README.md [hunk 15]
```diff
@@ -87,4 +95,4 @@ sudo apt install libopencv-imgcodecs3.4
 Is there any other version available?
 
 You can find nightly release version from [here](https://repository.apache.org/#nexus-search;gav~org.apache.mxnet~~1.5.0-SNAPSHOT~~).
-Please keep the same version in the Makefile or [above version](https://repository.apache.org/#nexus-search;gav~org.apache.mxnet~~~~) to run this demo.
+Please keep the same version in the pom file or [above version](https://repository.apache.org/#nexus-search;gav~org.apache.mxnet~~~~) to run this demo.
```
Comment (aaronmarkham) [inline scala-package/mxnet-demo/java-demo/README.md]: This is kind of weird to link to above version. If you want them to look at the above version then don't link. Offer the link as a backup resource where they can see available versions...

File: scala-package/mxnet-demo/scala-demo/README.md [hunk 16]
```diff
@@ -1,7 +1,15 @@
 # MXNet Scala Sample Project
 This is an project created to use Maven-published Scala package with two Scala examples.
 ## Setup
-User are required to use `mvn package` to build the package,
+User are required to use maven to build the package, by simply running
```
Comment (aaronmarkham) [inline scala-package/mxnet-demo/scala-demo/README.md]:
```suggestion
You are required to use maven to build the package, by running the following:
```
Again, is it really a requirement or one route to do things?

File: scala-package/mxnet-demo/scala-demo/README.md [hunk 17]
```diff
@@ -54,9 +55,9 @@ If you want to test run on GPU, you can set a environment variable as follows:
 export SCALA_TEST_ON_GPU=1
 ```
 ## Clean up
-Clean up for Maven package is simple, you can run the pre-configed `Makefile` as:
+Clean up for Maven package is simple, you can just run
```
Comment (aaronmarkham) [inline scala-package/mxnet-demo/scala-demo/README.md]:
```suggestion
To clean up a Maven package, run the following:
```

File: docs/install/java_setup.md (hunk 7 above)
Comment (lanking520) [reply docs/install/java_setup.md]: This is known, mvn command could only execute in a folder that has a pom file.

File: scala-package/mxnet-demo/java-demo/README.md (hunk 14 above)
Comment (lanking520) [reply scala-package/mxnet-demo/java-demo/README.md]: This is specifically for users trying to use Eclipse in their demo project

File: docs/install/java_setup.md (hunk 7 above)
Comment (lanking520) [reply docs/install/java_setup.md]: I also add one line to explain that

**Comment by lanking520** [comment]
@aaronmarkham please give a 2nd phase review
